lib/models/url_preferences_model.dart

```diff
@@ -1,18 +1,28 @@
-class UrlPreferences {
-  List<String> favouriteWidgetIds = [];
-  List<String> quarantineWidgetIds = [];
+import 'dart:convert';
 
-  UrlPreferences({this.favouriteWidgetIds, this.quarantineWidgetIds});
+import 'package:cogboardmobileapp/models/widget_model.dart';
 
-  factory UrlPreferences.fromJson(Map<String, dynamic> json) => UrlPreferences(
-        favouriteWidgetIds: (json['favouriteWidgetIds'] as List<dynamic>).cast<String>(),
-        quarantineWidgetIds: (json['quarantineWidgetIds'] as List<dynamic>).cast<String>(),
+class ConnectionPreferences {
+  List<DashboardWidget> favouriteWidgets = [];
+  List<DashboardWidget> quarantineWidgets = [];
+  String connectionName;
+  String connectionUrl;
+
+  ConnectionPreferences({this.favouriteWidgets, this.quarantineWidgets, this.connectionName, this.connectionUrl});
+
+  factory ConnectionPreferences.fromJson(Map<String, dynamic> json) => ConnectionPreferences(
+    favouriteWidgets: List<dynamic>.from(jsonDecode(json['favouriteWidgets'])).map((e) => DashboardWidget.fromJson(e)).toList(),
+    quarantineWidgets: List<dynamic>.from(jsonDecode(json['quarantineWidgets'])).map((e) => DashboardWidget.fromJson(e)).toList(),
+    connectionName: json['connectionName'],
+    connectionUrl: json['connectionUrl'],
       );
 
   Map<String, dynamic> toJson() {
     return {
-      'favouriteWidgetIds': favouriteWidgetIds,
-      'quarantineWidgetIds': quarantineWidgetIds,
+      'favouriteWidgets': jsonEncode(favouriteWidgets.map((i) => i.toJson()).toList()).toString(),
```

Comment:
[Later] extract method for: `jsonEncode(favouriteWidgets.map((i) => i.toJson()).toList()).toString()`

Reply:
done